plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceDescriptor.java

```diff
@@ -322,6 +325,9 @@ DBPDataSourceOrigin getOriginSource() {
 
     public void setDriver(@NotNull DriverDescriptor driver) {
         this.driver = driver;
+        if (this.driver.isSingleConnection()) {
+            this.setForceUseSingleConnection(true);
+        }
```

Comment:
Not sure about adding this kind of interaction inside setter

Reply:
I'd prefer to replace it with 
`this.forceUseSingleConnection = driver.isSingleConnection();`